### ISSUE TITLE
IllegalStateException: IBinder used after unbindService 

### ISSUE THREAD

**Issue by Agraphie** [amplitude/Amplitude-Android#153]
Hi there,

I got the following exception in our production App. I am not sure how to reproduce it, as it was not discovered by me. We are using version 2.15.0

```Java
Exception java.lang.IllegalStateException: IBinder used after unbindService
android.os.Parcel.readException (Parcel.java:1701)
android.os.Parcel.readException (Parcel.java:1646)
android.location.ILocationManager$Stub$Proxy.getFromLocation (ILocationManager.java:875)
android.location.Geocoder.getFromLocation (Geocoder.java:133)
com.amplitude.api.DeviceInfo$CachedInfo.getCountryFromLocation (DeviceInfo.java:141)
com.amplitude.api.DeviceInfo$CachedInfo.getCountry (DeviceInfo.java:119)
com.amplitude.api.DeviceInfo$CachedInfo.<init> (DeviceInfo.java:64)
com.amplitude.api.DeviceInfo$CachedInfo.<init> (DeviceInfo.java:41)
com.amplitude.api.DeviceInfo.getCachedInfo (DeviceInfo.java:266)
com.amplitude.api.DeviceInfo.prefetch (DeviceInfo.java:272)
com.amplitude.api.AmplitudeClient.initializeDeviceInfo (AmplitudeClient.java:305)
com.amplitude.api.AmplitudeClient.access$000 (AmplitudeClient.java:39)
com.amplitude.api.AmplitudeClient.access$102 (AmplitudeClient.java:39)
com.amplitude.api.AmplitudeClient.access$200 (AmplitudeClient.java:39)
com.amplitude.api.AmplitudeClient.access$1002 (AmplitudeClient.java:39)
com.amplitude.api.AmplitudeClient$1.run (AmplitudeClient.java:242)
android.os.Handler.handleCallback (Handler.java:751)
android.os.Handler.dispatchMessage (Handler.java:95)
android.os.Looper.loop (Looper.java:154)
android.os.HandlerThread.run (HandlerThread.java:61)
```

Is this a known issue? I could not find anything about it.

**Comment by djih** [comment]
@Agraphie thanks for reporting. Can you provide any more information? How often are you seeing this exception / on how many users or devices?

**Comment by Agraphie** [comment]
Thanks for replying @djih. Unfortunately, I do not have much more information. It only appeared once so far on a Samsung SM-G955F with API level 24.

**Comment by djih** [comment]
@Agraphie doesn't seem to be a widespread issue. What I can do is catch the exception so it doesn't crash the sdk and your app: https://github.com/amplitude/Amplitude-Android/pull/155. Expect a patch to go out either today or tomorrow

**Comment by Agraphie** [comment]
Thanks for that @djih. I will update this issue in case I get any more stacktraces or any more information.

**Comment by djih** [comment]
@Agraphie we have released v2.16.0 which should handle the exception: https://github.com/amplitude/Amplitude-Android/releases/tag/v2.16.0

**Comment by Agraphie** [comment]
Thanks @djih, appreciate the quick fix!
